pkgs/development/libraries/faad2/default.nix

```diff
@@ -13,19 +13,42 @@
 
 stdenv.mkDerivation rec {
   pname = "faad2";
-  version = "2.10.1";
+  version = "2.11.0";
 
   src = fetchFromGitHub {
     owner = "knik0";
     repo = "faad2";
     rev = version;
-    sha256 = "sha256-k7y12OwCn3YkNZY9Ov5Y9EQtlrZh6oFUzM27JDR960w=";
+    hash = "sha256-fp8RGG9HQOcyudTA9osGaI3oVIY8vo8vhJtd04+wNKc=";
   };
 
-  configureFlags = []
-    ++ lib.optional drmSupport "--with-drm";
+  patches = [
+    # i686Linux packports
+    # TODO: remove with next release
+    (fetchpatch {
+      name = "lrintf-p1.patch";
+      url = "https://github.com/knik0/faad2/commit/1001f9576cbb29242671c489cd861de61cfe08e2.patch";
+      hash = "sha256-du3oIbtaSU0ZhzMyMmIBHA6lndImA0tmT5D8pK8r49g=";
+    })
+    (fetchpatch {
+      name = "lrintf-p2.patch";
+      url = "https://github.com/knik0/faad2/commit/047fd22172a7ff6974b9ac6ca7e4ffdb1944f2e0.patch";
+      hash = "sha256-F7VXjtiDbii2oBKuJHZbKSsYGyO+rolPk8KjcvPIRsw=";
+    })
+    (fetchpatch {
+      name = "lrintf-p3.patch";
+      url = "https://github.com/knik0/faad2/commit/2c1eebc89dd21ab25c24b664762515af39bd9700.patch";
+      hash = "sha256-MSd7dU5IxxpkqCGs1vTvMtOMz6CnY2nXt+w4mj0yd60=";
+    })
+  ];
 
-  nativeBuildInputs = [ autoreconfHook ];
+  outputs = [ "out" "dev" ];
+
+  nativeBuildInputs = [ cmake ];
+
+  cmakeFlags = lib.optionals (!stdenv.hostPlatform.isStatic) [
```

Comment:
just curious, is this check needed? I thought [`makeStaticLibraries` set cmakeFlags](https://github.com/NixOS/nixpkgs/blob/9290bb56546e4c9b99f7960ce08943a10108d65a/pkgs/stdenv/adapters.nix#L88) for static builds

Reply:
In this case it's the non-static build that needs it (the inverted case). Otherwise `faad2` does not build shared library.

Reply:
I mean, could `cmakeFlags = [ "-DBUILD_SHARED_LIBS=ON" ]` be enough? since makeStaticLibraries would override it?